src/test/java/core/basesyntax/service/imp/TransactionParserImpTest.java

```diff
@@ -0,0 +1,85 @@
+package core.basesyntax.service.imp;
+
+import core.basesyntax.model.ProductTransaction;
+import core.basesyntax.service.TransactionParser;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TransactionParserImpTest {
+    private static final String CORRECT_LINE_BALANCE = "b,apple,20";
+    private static final String CORRECT_LINE_SUPPLY = "s,apple,20";
+    private static final String CORRECT_LINE_PURCHASE = "p,apple,20";
+    private static final String CORRECT_LINE_RETURN = "r,apple,20";
+    private static final String INVALID_TYPE_LINE = "j,apple,20";
+    private static final String ZERO_QUANTITY_LINE = "b,apple,0";
+    private static final String INVALID_RECORDS = "banana, 0";
+    private static final String APPLE = "apple";
+    private static final int QUANTITY = 20;
+    private static TransactionParser transactionParser;
+
+    @BeforeClass
+    public static void beforeClass() {
+        transactionParser = new TransactionParserImp();
+    }
+
+    @Test
+    public void parseValidBalanceFull_ok() {
+        ProductTransaction transaction = transactionParser.parse(CORRECT_LINE_BALANCE);
+        Assert.assertEquals("Product type: ", ProductTransaction.Operation.BALANCE,
+                transaction.getOperation());
+        Assert.assertEquals("Product name: ", APPLE, transaction.getProduct());
+        Assert.assertEquals("Product quantity: ", QUANTITY, transaction.getQuantity());
+    }
+
+    @Test
+    public void parseValidSupply_ok() {
+        ProductTransaction transaction = transactionParser.parse(CORRECT_LINE_SUPPLY);
+        Assert.assertEquals("Product type: ", ProductTransaction.Operation.SUPPLY,
+                transaction.getOperation());
+    }
+
+    @Test
+    public void parseValidPurchase_ok() {
+        ProductTransaction transaction = transactionParser.parse(CORRECT_LINE_PURCHASE);
+        Assert.assertEquals("Product type: ", ProductTransaction.Operation.PURCHASE,
+                transaction.getOperation());
+    }
+
+    @Test
+    public void parseValidReturn_ok() {
+        ProductTransaction transaction = transactionParser.parse(CORRECT_LINE_RETURN);
+        Assert.assertEquals("Product type: ", ProductTransaction.Operation.RETURN,
+                transaction.getOperation());
+    }
+
+    @Test
+    public void parseAllValidInput_ok() {

```

Comment:
Update each test name according to the readme suggestion
```suggestion
    public void parse_validRecords_ok() {
```
it will increase readability a bit